app/controllers/species_lists_controller.rb

```diff
@@ -42,22 +42,25 @@ class SpeciesListsController < ApplicationController
     :update
   ]
 
-  def index # rubocop:disable Metrics/AbcSize
-    if params[:advanced_search].present?
-      advanced_search
-    elsif params[:pattern].present?
-      species_list_search
-    elsif params[:by_user].present?
-      species_lists_by_user
-    elsif params[:for_project].present?
-      species_lists_for_project
-    elsif params[:by] == "title"
-      species_lists_by_title
-    elsif params[:by].present?
-      index_species_list
-    else
-      list_species_lists
-    end
+  @index_subaction_param_keys = [
+    :pattern,
+    :by_user,
+    :for_project,
+    :by
+  ].freeze
+
+  @index_subaction_dispatch_table = {
+    pattern: :species_list_search,
+    by_user: :species_lists_by_user,
+    for_project: :species_lists_for_project,
+    by: :by_title_or_selected_by_query
```

Comment:
Dispatches to subaction which merely conditionally switches to different subactions.